NullPointerException when generating the report

### Expected behavior
Generate the report
### Actual behavior
{
"errors": [
{
"msg": "An error has occurred. Please contact your administrator"
}
]
}
### Steps to reproduce behavior
I'm using docker with 7.9 LTS sonarqube, after selecting the project and clicking the generate button i get that api response instead of the file.
### Detection version
3.0.0 plugin mode

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project cnesreport: There are test failures.
[ERROR]
[ERROR] Please refer to /home/desaadmin/sonar-cnes-report/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/desaadmin/sonar-cnes-report && /usr/lib/jvm/java-11-openjdk-amd64/bin/java -javaagent:/home/desaadmin/.m2/repository/org/jacoco                                                                    /org.jacoco.agent/0.8.0/org.jacoco.agent-0.8.0-runtime.jar=destfile=/home/desaadmin/sonar-cnes-report/target/jacoco.exec -jar /home/desaadmin/sonar-cnes-report/target/                                                                    surefire/surefirebooter12709292607794703493.jar /home/desaadmin/sonar-cnes-report/target/surefire 2019-08-22T12-12-58_994-jvmRun1 surefire13227860691335322921tmp suref                                                                    ire_02725595737273684419tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/desaadmin/sonar-cnes-report && /usr/lib/jvm/java-11-openjdk-amd64/bin/java -javaagent:/home/desaadmin/.m2/repository/org/jacoco                                                                    /org.jacoco.agent/0.8.0/org.jacoco.agent-0.8.0-runtime.jar=destfile=/home/desaadmin/sonar-cnes-report/target/jacoco.exec -jar /home/desaadmin/sonar-cnes-report/target/                                                                    surefire/surefirebooter12709292607794703493.jar /home/desaadmin/sonar-cnes-report/target/surefire 2019-08-22T12-12-58_994-jvmRun1 surefire13227860691335322921tmp suref                                                                    ire_02725595737273684419tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:670)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:283)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1161)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1002)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:848)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)

I got this when building the pluging

--- a/.travis.yml
+++ b/.travis.yml
@@ -58,7 +58,7 @@ deploy:
   provider: releases
   api_key:
     secure: PFc97cepts1tQHmAWMIoLUVFo3aP6o6sVVg6B+30YjguyO6SrnFf/uE+zem1ogP6S3NOcMY0xtfBM4QSHoOTWVp1C5nk7SPzvXUs/diP5J7B58Z8OhMl0vkjz2Xa4GQx6ndK07DgMUeJyxexTBatJJ47E1Y+xLHqkSbcCCALfD4LfRvAM902t9nHBcRUPONmbAX6fj2z5TRTc9AveWwwgAbfHPjsh2WvuEQ0bSttKIYO08rm4GfXiqcwcfUtBwRLij+tQEYmyj+D5ie1jfJZe61FEF9rH1bS44ARr3BMqm1Ezm6rIEYG+4f1h5xZnXjX50uYIlShH9me28+XPr7FCopOwJyim/cT4Lviypw6dki4FRtRig93+Gjj42rLXSQaL4VtA1p+KscCBrG3C4QCbIOEixtN8n/WDd1U90x3GX8X+iPxApw5pWJ/Svc7ggjh9f0WvpsFGlFowqQBsP0GrGwdDoNeu5qaHA6G1AIHses6EUcq/WZ6ZAy/cbQCNjICPIzXYSV8pBCKjnnXZWZfciZWH3Q/EghCnyyjCGLpYV2cEtHgGx50rGWQVyPHmKZZ6/B2CETWPtxpg//yXPRsTII0ssBbgFZgmeiW6Hl+TQv6AEoLK4YJhPod8XEenYKwanct2zGbOH6YH6emM0vE3HN3ldfHNRHtyhPyGsefsc0=
-  file: sonar-cnes-report.jar
+  file: target/sonar-cnes-report.jar
   skip_cleanup: true
   draft: true
   on:

--- a/README.md
+++ b/README.md
@@ -113,7 +113,7 @@ java -Dhttps.proxyHost=https://myproxy -Dhttps.proxyPort=42
   <td><b>2.0.0<br/>Standalone only</b></td>
   <td><b>2.1.0<br/>Standalone only</b></b></td>
   <td><b>2.2.0<br/>Standalone + Plugin</b></td>
-  <td><b>3.0.0<br/>Standalone + Plugin</b></b></td>
+  <td><b>3.0.x<br/>Standalone + Plugin</b></b></td>
  </tr>
  <tr>
   <td><b>3.7.x (LTS)</b></td>

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>fr.cnes.sonar</groupId>
     <artifactId>cnesreport</artifactId>
-    <version>3.0.0</version>
+    <version>3.0.1</version>
     <packaging>sonar-plugin</packaging>
 
     <name>SonarQube CNES Report</name>

--- a/src/main/java/fr/cnes/sonar/report/exporters/docx/DataAdapter.java
+++ b/src/main/java/fr/cnes/sonar/report/exporters/docx/DataAdapter.java
@@ -679,20 +679,28 @@ class RuleComparator implements Comparator<String>{
     }
 
     public int compare(String o1, String o2) {
-        if(o1.isEmpty() || o2.isEmpty())return 0;
+        int compare;
 
-        int compare = report
-                .getRule(o1)
-                .getType()
-                .compareTo(
-                report.getRule(o2).getType()
-        );
-        if (compare == 0) compare = report.getRule(o1).getSeverity().compareTo(
-                report.getRule(o2).getSeverity()
-        );
-        if (compare == 0) compare = report.getRule(o1).getKey().compareTo(
-                report.getRule(o2).getKey()
-        );
+        if(o1.isEmpty() || o2.isEmpty())return 0;
+        if(report.getRule(o1) == null || report.getRule(o1) == null)return 0;
+
+        try {
+            compare = report
+                    .getRule(o1)
+                    .getType()
+                    .compareTo(
+                            report.getRule(o2).getType()
+                    );
+            if (compare == 0) compare = report.getRule(o1).getSeverity().compareTo(
+                    report.getRule(o2).getSeverity()
+            );
+            if (compare == 0) compare = report.getRule(o1).getKey().compareTo(
+                    report.getRule(o2).getKey()
+            );
+        }
+        catch (NullPointerException e){
+            compare = 0;
+        }
 
         return compare;
     }

--- a/src/main/java/fr/cnes/sonar/report/exporters/docx/DataAdapter.java
+++ b/src/main/java/fr/cnes/sonar/report/exporters/docx/DataAdapter.java
@@ -247,7 +247,7 @@ public final class DataAdapter {
     /**
      * List of possible issue types
      */
-    private static final String[] ISSUE_TYPES = {"VULNERABILITY", "BUG", "CODE_SMELL"};
+    private static final String[] ISSUE_TYPES = {"VULNERABILITY", "BUG", "CODE_SMELL", "SECURITY_HOTSPOT"};
     /**
      * List of possible issue severities
      */
@@ -669,7 +669,7 @@ public final class DataAdapter {
 }
 
 /**
- * RuleComparator is used to compare 2 issues to sort them by severityq
+ * RuleComparator is used to compare 2 issues to sort them by type & severity
  */
 class RuleComparator implements Comparator<String>{
     Report report;
@@ -679,27 +679,33 @@ class RuleComparator implements Comparator<String>{
     }
 
     public int compare(String o1, String o2) {
-        int compare;
+        int compare = 0;
 
-        if(o1.isEmpty() || o2.isEmpty())return 0;
-        if(report.getRule(o1) == null || report.getRule(o1) == null)return 0;
+        if(o1.isEmpty() || o2.isEmpty())compare = 1;
 
-        try {
+        //If rule is removed in quality gate, the issue is send to the end of list
+        if(report.getRule(o1) == null){compare = 1;}
+        else if(report.getRule(o2) == null){compare = -1;}
+
+        if (compare == 0){
             compare = report
                     .getRule(o1)
                     .getType()
                     .compareTo(
                             report.getRule(o2).getType()
                     );
-            if (compare == 0) compare = report.getRule(o1).getSeverity().compareTo(
+        }
+
+        if (compare == 0) {
+            compare = report.getRule(o1).getSeverity().compareTo(
                     report.getRule(o2).getSeverity()
             );
-            if (compare == 0) compare = report.getRule(o1).getKey().compareTo(
+        }
+
+        if (compare == 0){
+            compare = report.getRule(o1).getKey().compareTo(
                     report.getRule(o2).getKey()
             );
-        }
-        catch (NullPointerException e){
-            compare = 0;
         }
 
         return compare;

--- a/src/main/java/fr/cnes/sonar/report/factory/ServerFactory.java
+++ b/src/main/java/fr/cnes/sonar/report/factory/ServerFactory.java
@@ -32,7 +32,7 @@ public class ServerFactory {
 
     /** List of SonarQube versions which are supported by cnesreport. */
     private static final List<String> SUPPORTED_VERSIONS = Arrays.asList(
-            "5.6.*", "6.*", "7.*");
+            "6.6", "6.6.*", "6.7", "6.7.*", "7.*", "8.*");
 
     /** Url of the SonarQube server. */
     private String url;

